crates/net/downloaders/src/bodies/concurrent.rs

```diff
@@ -131,41 +131,39 @@ where
             return Ok(headers.into_iter().cloned().map(BlockResponse::Empty).collect())
         }
 
-        let request_len = headers_with_txs_and_ommers.len();
-        tracing::trace!(target: "downloaders::bodies", request_len, "Requesting bodies");
-        let (peer_id, bodies) =
-            self.client.get_block_bodies(headers_with_txs_and_ommers).await?.split();
-        tracing::trace!(
-            target: "downloaders::bodies", request_len, response_len = bodies.len(), ?peer_id, "Received bodies"
-        );
+        let mut responses = Vec::with_capacity(headers_with_txs_and_ommers.len());
+        while responses.len() != headers_with_txs_and_ommers.len() {
```

Comment:
should add a note on why we do this so when we handroll the stream we remember

Reply:
comment added